akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -29,23 +29,13 @@ class Encoder(
 
       private val noSender = transport.system.deadLetters.path.toSerializationFormatWithAddress(localAddress)
       private val senderCache = new java.util.HashMap[ActorRef, String]
-      private var recipientCache = new java.util.HashMap[ActorRef, String]
 
       override def onPush(): Unit = {
         val send = grab(in)
         val envelope = pool.acquire()
 
-        val recipientStr = recipientCache.get(send.recipient) match {
-          case null ⇒
-            val s = send.recipient.path.toSerializationFormat
-            // FIXME this cache will be replaced by compression table
-            if (recipientCache.size() >= 1000)
-              recipientCache.clear()
-            recipientCache.put(send.recipient, s)
-            s
-          case s ⇒ s
-        }
-        headerBuilder.recipientActorRef = recipientStr
+        // internally compression is applied by the builder:
+        headerBuilder.recipientActorRef = send.recipient.path.toSerializationFormat
```

Comment:
this is also something we must optimize, `toSerializationFormat` is way to costly, perhaps also something we can cache in the RemoteActorRef